### ISSUE TITLE
Datagrid column resizer does not have associated visible label

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4767]
Reported by @kristastarr in the [Datagrid review](https://github.ibm.com/CDAI-design/pal/issues/1575) from the accessibility audit.

> The column resizer inputs / sliders don't have an associated visible label

This is a flagged as a _potential violation_. There’s already an `aria-label` on the element, but not a separate associated label in the DOM.

```html
<input
  style="cursor: col-resize;"
  value="106"
  aria-label="Resize column"
  type="range"
  class="c4p--datagrid__col-resizer-range"
  draggable="false">
```

![image](https://github.com/carbon-design-system/ibm-products/assets/1717398/4efe343c-3b7c-4742-ab9e-6cb605c4d936)

> ### What to do
> If there is already a visible label for the <input> element, use the for attribute on the label to reference the <input> element's id value
> **Or**, add a visible label, with the for attribute linking it to the <input>
> 
> Rule ID: input_label_visible
> Reason ID: Potential_1
> 
> ### About this requirement
> [IBM 3.3.2 Label or Instruction](https://www.ibm.com/able/requirements/requirements/#3_3_2)
> [IBM 2.5.3 Label in Name](https://www.ibm.com/able/requirements/requirements/#2_5_3)
> [H44: Associate text labels with form controls](https://www.w3.org/WAI/WCAG22/Techniques/html/H44)
> 
> ### Who does this affect?
> People with dexterity impairments using voice control
